src/apps/workspaces/service/user_applet_access.py

```diff
@@ -183,17 +185,20 @@ async def add_role_by_private_invitation(self, role: Role):
             )
         else:
             meta = dict()
-        access_schema = await UserAppletAccessCRUD(self.session).save(
-            UserAppletAccessSchema(
-                user_id=self._user_id,
-                applet_id=self._applet_id,
-                role=role,
-                owner_id=owner_access.user_id,
-                invitor_id=owner_access.user_id,
-                meta=meta,
-            )
+
+        schema = UserAppletAccessSchema(
+            user_id=self._user_id,
+            applet_id=self._applet_id,
+            role=role,
+            owner_id=owner_access.user_id,
+            invitor_id=owner_access.user_id,
+            meta=meta,
+            is_deleted=False,
+        )
+        await UserAppletAccessCRUD(self.session).upsert_user_applet_access(
+            schema,
+            where=UserAppletAccessSchema.is_deleted == true(),
```

Comment:
Please use hybrid method .soft_exists(False)

Reply:
It updated.